src/hipscat/pixel_math/margin_bounding.py

```diff
@@ -0,0 +1,263 @@
+"""Utilities to build bounding boxes around healpixels that include a neighor margin."""
+
+import healpy as hp
+import numpy as np
+
+from astropy.coordinates import SkyCoord
+from regions import PixCoord, PolygonSkyRegion
+import astropy.wcs as world_coordinate_system
+import astropy.units as u
+
+from . import pixel_margins as pm
+
+
+def get_margin_scale(pixel_order, margin_threshold=0.1):
+    """Get the scale value need to expand the pixel bounding box to include the `margin_threshold`.
+
+    Args:
+        pixel_order (int): the order of the pixel to which we're calculating the margin region for.
+        margin_threshold (float): the size of the border region in degrees.
+    Returns:
+        a float representing the scale factor.
+    """
+    if margin_threshold <= 0.0:
+        raise ValueError("margin_threshold must be greater than 0.")
+
+    resolution = hp.nside2resol(2**pixel_order, arcmin=True) / 60.0
+    scale = 1 + ((2 * margin_threshold) / resolution)
+    return scale
+
+
+def get_margin_bounds_and_wcs(pixel_order, pix, scale, step=10):
+    """Get the astropy `regions.PolygonPixelRegion` and `astropy.wcs` objects
+        for a given margin bounding box scale.
+
+    Args:
+        pixel_order (int): the order of the pixel to which we're calculating the margin region for.
+        pix (int): the healpixel which we're calculating the margin region for.
+        scale (float): the scale factor to apply to the given healpixel.
+        step (int): the amount of samples of one side of the given healpixel's boundaries.
+            total samples taken = 4 * step.
+    Returns:
+        polygons (list of tuples): a list of :obj:`regions.PolygonPixelRegion`
+            and :obj:`wcs.WCS` tuples, covering the full area of the given healpixels
+            scaled up area.
+    """
```

Comment:
Just one last thing, sorry! Could we add a line in this docstring warning that it doesn't work for pixels at the pole?

Reply:
good call, added a line to `check_margin_bounds` as well!